### DISCUSSION TITLE
[week16] hw2 相關疑問

### DISCUSSION THREAD

**Discussion by hsujoey** [Lidemy/forum#181] [前端相關問題]
hw2 題目如下

``` js
for(var i=0; i<5; i++) {
  console.log('i: ' + i)
  setTimeout(() => {
    console.log(i)
  }, i * 1000)
}
```

會輸出以下
```
i: 0
i: 1
i: 2
i: 3
i: 4
5
5
5
5
5
```

如果今天 setTimeout 設定的秒數都一樣，等於會同時輸出 5 個 5 ，此時該怎麼證明：輸出的第一個 5 ，是 for 迴圈第一圈時產生的 5？

這問題應該也可以換一個問法，也就是：該怎麼證明 Callback Queue 是先進先出的？

我想透過改寫題目的程式碼，來證明 此 5 非彼 5 這件事，但不知道怎麼下手，可以請大大提點一下嗎？ 

謝謝各位～ 
（如果我問題敘述不清楚，歡迎提問，我再補充！）

**Comment by cwc329** [comment]
這邊把 var 變成 let 應該就可以看出 callback queue 是先進先出的了。

**Reply by hsujoey** [reply]
謝謝助教！

**Reply by cwc329** [reply]
那你知道為何改成 let 就可以嗎？
